Clean-up Graph grouping after release

albertoblaz opened the pull request:
## Summary

Address feedback from merged PRs and fix issues that were not part of v9.2.0 release.

### Changes

Polishing
- [x] Replace string literals with `DOCUMENT_TYPE_{ENTITY|EVENT|ALERT}` constants
- [x] Address Alex's comments in merged PRs

Fixes
- [x] Only allow to expand details for single/grouped nodes when possible
- [x] i18n issues in `PaginationControls.tsx`

Enhancements
- [x] Navigate to event/alert preview panel from group preview panel (not possible for entities so link remains bogus for them)

Testing
- [x] Clean-up helper function in API FTR test
- [x] Write preview panel unit tests
- [x] Re-enable skipped GraphVisualization tests
- [x] Extend GraphVisualization tests to have coverage for grouped nodes
- [x] Re-enable skipped GraphInvestigation stories tests - had to skip because of bug expanding details for single/grouped nodes
- [x] Write e2e tests that test opening/closing preview panel and loading paginated document details

### How to test

1. Run Elasticsearch with `yarn es snapshot --license trial -E reindex.remote.whitelist=keep-long-live-env-ess.es.us-west2.gcp.elastic-cloud.com:443`
2. Run Kibana locally with `yarn start --no-base-path`
3. Install Cloud Asset Discovery (skip agent installation)
4. Re-index Asset Inventory data with this query
    ```
    POST _reindex?wait_for_completion=true
    {
      "conflicts": "proceed", 
      "source": {
        "remote": {
          "host": "${ES_REMOTE_HOST}", # <-- Set var in Config tab to https://keep-long-live-env-ess.es.us-west2.gcp.elastic-cloud.com:443
          "username": "${ES_REMOTE_USER}", # <-- Set var in Config tab to your user in https://keep-long-live-env-ess.kb.us-west2.gcp.elastic-cloud.com/app/management/security/users i.e. alberto
          "password": "${ES_REMOTE_PASS}" # <-- Set var in Config tab to your password in the same cloud env
        },
        "index": "logs-cloud_asset_inventory*",
        "query": {
          "bool": {
            "must": [
              {
                "range": {
                  "@timestamp": {
                    "gte": "now-1d"
                  }
                }
              }
            ]
          }
        }
      },
      "dest": {
        "op_type": "create",
        "index": "logs-cloud_asset_inventory.asset_inventory-default"
      }
    }
    ```
5. Reindex AWS Cloudtrail logs with this query
    ```
    POST _reindex?wait_for_completion=false
    {
      "conflicts": "proceed", 
      "source": {
        "remote": {
          "host": "${ES_REMOTE_HOST}",
          "username": "${ES_REMOTE_USER}",
          "password": "${ES_REMOTE_PASS}"
        },
        "index": "logs-aws.cloudtrail-*",
        "query": {
           "bool": {
             "must": [],
             "filter": [
               {
                 "range": {
                   "@timestamp": {
                     "gte": "now-3h",
                     "lte": "now"
                   }
                 }
               }
             ]
           }
         }
      },
      "dest": {
        "op_type": "create",
        "index": "logs-aws.cloudtrail-reindexed"
      }
    }
    ```
6. Create a new Data View that matches this index pattern `logs-aws.cloudtrail-*`. Give it the exact same name
7. Create a new rule associated with the Data View created in the previous step. Set the query to existing `@timestamp: * and target.entity.id: *`. Enable the rule.
8. Go to Alerts page. Set the date range from several days ago until now. Wait for alerts to show.
9. Open one alert flyout and check you can expand the Graph Visualization tab.
10. Click on all options to show related events, related entities, show actions on this entity, etc.... try to show as many nodes as possible and check they're grouped.

### Checklist

- [x] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)
- [ ] [Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html) was added for features that require explanation or tutorials
- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios
- [ ] If a plugin configuration key changed, check if it needs to be allowlisted in the cloud and added to the [docker list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [x] This was checked for breaking HTTP API changes, and any breaking changes have been approved by the breaking-change committee. The `release_note:breaking` label should be applied in these situations.
- [ ] [Flaky Test Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was used on any tests changed
- [x] The PR  description includes the appropriate Release Notes section, and the correct `release_note:*` label is applied per the [guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)
- [x] Review the [backport guidelines](https://docs.google.com/document/d/1VyN5k91e5OVumlc0Gb9RPa3h1ewuPE705nRtioPiTvY/edit?usp=sharing) and apply applicable `backport:*` labels.

### Identify risks

No risks for customers with feature hidden behind advanced setting.

albertoblaz committed:
Replace document-type literals with constants

albertoblaz committed:
Fix showing node details logic for single/groups

albertoblaz committed:
Clean-up FTR test

albertoblaz committed:
Address Alex's feedback in previous PRs

albertoblaz committed:
Open flyout preview panel via pub-sub mechanism

albertoblaz committed:
Write unit tests for GraphGroupedNodePreviewPanel

albertoblaz committed:
Rename panel implementation

albertoblaz committed:
Re-enable skipped GraphVisualization tests

albertoblaz committed:
Re-enable GraphInvestigation stories tests

albertoblaz committed:
Extend FTR UI tests

albertoblaz commented on x-pack/solutions/security/packages/kbn-cloud-security-posture/common/schema/graph/v1.ts:
```diff
@@ -6,7 +6,6 @@
  */
 
 import { schema } from '@kbn/config-schema';
-import { ApiMessageCode } from '../../types/graph/v1';
```
Removed circular dependency between `schema/graph/v1.ts` and `types/graph/v1.ts`. 

This allows us to import all `DOCUMENT_TYPE_{ENTITY/EVENT/ALERT}` constants from other files

albertoblaz commented on x-pack/solutions/security/packages/kbn-cloud-security-posture/graph/src/components/graph_investigation/use_entity_node_expand_popover.ts:
```diff
@@ -91,7 +91,9 @@ export const useEntityNodeExpandPopover = (
-      const shouldDisableEntityDetailsListItem = !onShowEntityDetailsClick;
+      const shouldDisableEntityDetailsListItem =
+        !onShowEntityDetailsClick ||
+        !['single-entity', 'grouped-entities'].includes(getNodeDocumentMode(node.data));
```
Fix that disables "show entity details" action when needed

albertoblaz reviewed (commented):
Notes for reviewers down below

elasticmachine commented:
Pinging @elastic/kibana-cloud-security-posture (Team:Cloud Security)

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/354391)
* Commit: 873adab7b1702b93a5ed5aed966308950003e8b5
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-238395)
* Kibana Serverless Image: `docker.elastic.co/kibana-ci/kibana-serverless:pr-238395-873adab7b170`

### Failed CI Steps
* [Jest Tests #4](https://buildkite.com/elastic/kibana-pull-request/builds/354391#019a1582-8aab-4f33-9cae-bb116fb6af99)
* [Jest Tests #4](https://buildkite.com/elastic/kibana-pull-request/builds/354391#019a1597-3cb8-4ac6-a416-4a612a5d782f)
* [Jest Integration Tests #15](https://buildkite.com/elastic/kibana-pull-request/builds/354391#019a1582-8aba-4aee-a22a-fad6b39c369f)
* [FTR Configs #50](https://buildkite.com/elastic/kibana-pull-request/builds/354391#019a1582-8b01-4790-be35-9e348fc632c7)
* [FTR Configs #109](https://buildkite.com/elastic/kibana-pull-request/builds/354391#019a1582-8b4a-42ac-b92b-6b3389b83797)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/354391#019a1582-8aab-4f33-9cae-bb116fb6af99) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/354391/jobs/019a1582-8aab-4f33-9cae-bb116fb6af99/artifacts/019a1597-2552-4502-8cdc-ac46d424fa61) Jest Tests #<span></span>4 / <HeaderRow /> emits click event once for a single click
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/354391#019a1597-3cb8-4ac6-a416-4a612a5d782f) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/354391/jobs/019a1597-3cb8-4ac6-a416-4a612a5d782f/artifacts/019a15ab-eff3-4d1b-9efd-6466ed43f5e1) Jest Tests #<span></span>4 / <HeaderRow /> emits click event once for a single click
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/354391#019a1582-8aab-4f33-9cae-bb116fb6af99) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/354391/jobs/019a1582-8aab-4f33-9cae-bb116fb6af99/artifacts/019a1597-2552-4648-a534-f214db29a2c3) Jest Tests #<span></span>4 / <HeaderRow /> suppresses rapid duplicate clicks within dedupe window
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/354391#019a1597-3cb8-4ac6-a416-4a612a5d782f) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/354391/jobs/019a1597-3cb8-4ac6-a416-4a612a5d782f/artifacts/019a15ab-eff2-4314-8e0d-19e462ca00bc) Jest Tests #<span></span>4 / <HeaderRow /> suppresses rapid duplicate clicks within dedupe window
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/354391#019a1582-8b4a-42ac-b92b-6b3389b83797) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/354391/jobs/019a1582-8b4a-42ac-b92b-6b3389b83797/artifacts/019a15aa-9777-4b43-ab50-d5864517f125) FTR Configs #<span></span>109 / discover/context_awareness extension getCellRenderers data view mode data source profile should not render custom @<span></span>timestamp or log.level
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/354391#019a1582-8aba-4aee-a22a-fad6b39c369f) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/354391/jobs/019a1582-8aba-4aee-a22a-fad6b39c369f/artifacts/019a1587-698d-4223-a844-830b8f8cf05e) Jest Integration Tests #<span></span>15 / workflow level timeout should invoke connector only 3 times timeout
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/354391#019a1582-8aba-4aee-a22a-fad6b39c369f) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/354391/jobs/019a1582-8aba-4aee-a22a-fad6b39c369f/artifacts/019a1587-698d-4050-a2b5-f9e883665e56) Jest Integration Tests #<span></span>15 / workflow with wait step when duration is short should wait for the specified duration between firstConnectorStep and lastConnectorStep

### History
* :broken_heart: [Build #351614](https://buildkite.com/elastic/kibana-pull-request/builds/351614) failed b29e033323edfc21c1d8fc03a7fb607ee76154d0
* :broken_heart: [Build #351478](https://buildkite.com/elastic/kibana-pull-request/builds/351478) failed a8ff5e4f76a64ea3912c7d9857c95ae0d956c1b2

cc @albertoblaz

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/354391","number":354391,"commit":"873adab7b1702b93a5ed5aed966308950003e8b5"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/351614","number":351614,"commit":"b29e033323edfc21c1d8fc03a7fb607ee76154d0"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/351478","number":351478,"commit":"a8ff5e4f76a64ea3912c7d9857c95ae0d956c1b2"}],"number":354391}
buildkite-pr-comment-->

albertoblaz committed:
Fix unit tests for pub-sub

albertoblaz committed:
Find by test subject, reuse test-id

albertoblaz commented:
@elasticmachine merge upstream

albertoblaz committed:
CR: Refactor repeated lines in test

albertoblaz committed:
CR: Make item.id required

PhilippeOberti reviewed (approved):
Code review only, LGTM for the @elastic/security-threat-hunting-investigations team

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/356530)
* Commit: 3f9e09ead6798dae895c57a60a497a9b9dae73ed
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-238395)
* Kibana Serverless Image: `docker.elastic.co/kibana-ci/kibana-serverless:pr-238395-3f9e09ead679`

### Failed CI Steps
* [FTR Configs #19](https://buildkite.com/elastic/kibana-pull-request/builds/356530#019a33ed-f5fb-4091-b245-1d827b0e466e)
* [FTR Configs #38](https://buildkite.com/elastic/kibana-pull-request/builds/356530#019a33ed-f60c-4f8c-90b6-06e69e07858f)
* [FTR Configs #68](https://buildkite.com/elastic/kibana-pull-request/builds/356530#019a33ed-f627-4dbd-9cca-b1be3748be0c)
* [FTR Configs #74](https://buildkite.com/elastic/kibana-pull-request/builds/356530#019a33ed-f62d-4bfb-b31a-7aa387d10b7a)
* [FTR Configs #77](https://buildkite.com/elastic/kibana-pull-request/builds/356530#019a33ed-f630-4211-a666-60e644240df6)
* [FTR Configs #98](https://buildkite.com/elastic/kibana-pull-request/builds/356530#019a33ed-f64a-44b8-bb25-9adcaa0888e2)
* [FTR Configs #100](https://buildkite.com/elastic/kibana-pull-request/builds/356530#019a33ed-f64d-42cb-98ce-7c06e038d801)
* [FTR Configs #113](https://buildkite.com/elastic/kibana-pull-request/builds/356530#019a33ed-f65a-4325-9a47-9bd77fdcf58c)
* [FTR Configs #121](https://buildkite.com/elastic/kibana-pull-request/builds/356530#019a33ed-f662-45e4-b472-c36c98ded836)
* [Scout: [ platform / discover_enhanced ] plugin](https://buildkite.com/elastic/kibana-pull-request/builds/356530#019a33f4-1eba-4263-bfcb-c9b3f441d2f6)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/356530#019a33ed-f662-45e4-b472-c36c98ded836) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/356530/jobs/019a33ed-f662-45e4-b472-c36c98ded836/artifacts/019a3410-27ef-4478-99f0-5fbc0734f079) FTR Configs #<span></span>121 / Cloud Security Posture Security Network Page - Graph visualization expanded flyout - show grouped event details
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/356530#019a33ed-f627-4dbd-9cca-b1be3748be0c) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/356530/jobs/019a33ed-f627-4dbd-9cca-b1be3748be0c/artifacts/019a3419-e3ab-40ea-a7f9-eac3ffe52434) FTR Configs #<span></span>68 / saved objects management apis find with kibana index - relationships `hasReference` and `hasReferenceOperator` parameters search for a reference
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/356530#019a33f4-1eba-4263-bfcb-c9b3f441d2f6) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/356530/jobs/019a33f4-1eba-4263-bfcb-c9b3f441d2f6/artifacts/019a340f-1955-4025-9026-8a6248707fa5) Scout: [ platform / discover_enhanced ] plugin / serverless-oblt - Discover app - saved searches - should unselect saved search when navigating to a 'new'

### History
* :broken_heart: [Build #356408](https://buildkite.com/elastic/kibana-pull-request/builds/356408) failed 0f4974f12f46c3916cdac6a45b3dd040cdfbdd81
* :broken_heart: [Build #356318](https://buildkite.com/elastic/kibana-pull-request/builds/356318) failed e7ee19bf7cd59fc304233fc4f5e2120ac100bf27
* :green_heart: [Build #354936](https://buildkite.com/elastic/kibana-pull-request/builds/354936) succeeded 66185a217428d9fc513b044187fbb33eb3957a57
* :yellow_heart: [Build #354551](https://buildkite.com/elastic/kibana-pull-request/builds/354551) was flaky db78a779963306f0c6dc4ef0f9b292aafdae5e77
* :broken_heart: [Build #354391](https://buildkite.com/elastic/kibana-pull-request/builds/354391) failed 873adab7b1702b93a5ed5aed966308950003e8b5

cc @albertoblaz

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/356530","number":356530,"commit":"3f9e09ead6798dae895c57a60a497a9b9dae73ed"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/356408","number":356408,"commit":"0f4974f12f46c3916cdac6a45b3dd040cdfbdd81"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/356318","number":356318,"commit":"e7ee19bf7cd59fc304233fc4f5e2120ac100bf27"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/354936","number":354936,"commit":"66185a217428d9fc513b044187fbb33eb3957a57"},{"buildStatus":{"state":"running","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/354551","number":354551,"commit":"db78a779963306f0c6dc4ef0f9b292aafdae5e77"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/354391","number":354391,"commit":"873adab7b1702b93a5ed5aed966308950003e8b5"}],"number":356530}
buildkite-pr-comment-->

kfirpeled commented:
Since this task is regarding cleanup I see it fit to comment on few UI nits

<details>
<summary>Some screenshots from storybook: 📸 </summary>


**Redundant tooltip?**
<img width="1914" height="318" alt="Screenshot 2025-11-02 at 17 37 42" src="https://github.com/user-attachments/assets/0d287181-0187-45cb-8362-f91e886dbca2" />
<img width="1756" height="370" alt="Screenshot 2025-11-02 at 17 37 56" src="https://github.com/user-attachments/assets/fc87efc0-2046-409d-84d1-ed0769936daa" />

**Back icon on the refresh CTA is redundant IMO** 
<img width="988" height="490" alt="Screenshot 2025-11-02 at 17 38 33" src="https://github.com/user-attachments/assets/88557544-132d-4349-970a-784327074b0d" />

**Grouped entities/Grouped events**

IMO, Title should be either Grouped entities or Grouped events. Depending on the use case
Sub title: Loading entities or Loading events

<img width="934" height="986" alt="Screenshot 2025-11-02 at 17 39 05" src="https://github.com/user-attachments/assets/35c6ac81-5738-4117-8836-665d3eb9f6b7" />

**Missing colon after `Geo`**

<img width="770" height="214" alt="Screenshot 2025-11-02 at 17 56 19" src="https://github.com/user-attachments/assets/2a48cf5d-f48e-43a0-b2cc-c09567fb3963" />

**Investigation's storybook is broken**

<img width="3452" height="1880" alt="Screenshot 2025-11-02 at 17 58 39" src="https://github.com/user-attachments/assets/d9aa5e11-3ebf-497a-9ad7-3b171fe24971" />

**Hover button is not centered when hovered**

<img width="518" height="228" alt="Screenshot 2025-11-02 at 17 59 23" src="https://github.com/user-attachments/assets/f1007ae8-0420-407a-8b98-5cffe9f98647" />

**High contrast of the background in borealis dark theme. I didn't expect to see difference from the light theme**

<img width="1002" height="1048" alt="Screenshot 2025-11-02 at 18 00 41" src="https://github.com/user-attachments/assets/ac85b92d-d678-48df-9513-9224aa0a62f7" />
<img width="1034" height="506" alt="Screenshot 2025-11-02 at 18 00 58" src="https://github.com/user-attachments/assets/cea86a9a-54a7-49fe-800f-7e12c2afab68" />

</details>

cc: @nick-alayil , @larissaoliveiraelastic

kfirpeled commented:
🐞  Issue:

1. Total amount of items is larger than actual amount disabled when grouped

https://github.com/user-attachments/assets/4e90c3f0-b12f-4716-8a90-ecee62a38692


✅  Verified: 

1. Disable `show entity details` when not available
<img width="982" height="544" alt="Screenshot 2025-11-02 at 18 21 08" src="https://github.com/user-attachments/assets/18a057db-3015-4f50-b70e-cff4c0c42352" />
2. Allows to view entity details from the grouped preview flyout

kfirpeled commented on x-pack/solutions/security/packages/kbn-cloud-security-posture/graph/src/components/graph_grouped_node_preview_panel/components/grouped_item/parts/header_row.test.tsx:
```diff
@@ -0,0 +1,51 @@
+import { GROUPED_ITEM_TITLE_TEST_ID } from '../../../test_ids';
+import { HeaderRow } from './header_row';
+
+const flushMicrotasks = () => new Promise((r) => setTimeout(r, 0));
```
can you document why it is needed?
What does it solve?

I haven't found any reference of this pattern so it worth adding some details

kfirpeled commented on x-pack/solutions/security/plugins/cloud_security_posture/server/routes/graph/fetch_graph.ts:
```diff
@@ -289,7 +289,7 @@ ${
   )
 | STATS badge = COUNT(*),
-  uniqueEventsCount = COUNT_DISTINCT(CASE(isAlert == false, event.id, null)),
+  totalEventsCount = COUNT_DISTINCT(event.id),
```
As for counting alerts - I'd count by the alert id/doc id - because you might have more than one alert per event.  

As for counting event.id. I'm not sure if it is correct to count both alerts and events. Can you explain the use case you were solving here?

albertoblaz commented on x-pack/solutions/security/packages/kbn-cloud-security-posture/graph/src/components/graph_grouped_node_preview_panel/components/grouped_item/grouped_item.tsx:
```diff
@@ -35,7 +39,7 @@ export const GroupedItem = memo(({ item, isLoading }: GroupedItemProps) => {
 
   return (
-    <Panel isAlert={item.itemType === 'alert'}>
+    <Panel isAlert={item.itemType === DOCUMENT_TYPE_ALERT}>
```
Alright, that sounds to me like it must be raised and run by Product & Design. Figma only shows a single IP and Geo value per Grouped item. 

So I'd say let's keep it for now. If we change the Figma, I'll show more values according to the new design or else we can write a safer implementation that only renders the first item in the array

albertoblaz committed:
Address Kfir's CR comments

albertoblaz commented:
@kfirpeled:

> **1. Redundant tooltip?** 

Removed

> **2. Back icon on the refresh CTA is redundant IMO** 

Removed for now. Anyhow, @larissaoliveiraelastic is working on enhanced version of this so I wouldn't spent time on more changes

<img width="506" height="306" alt="Screenshot 2025-11-03 at 15 05 30" src="https://github.com/user-attachments/assets/33b59370-0657-49a5-aed8-91e3cec0389f" />
 
> **3. Grouped entities/Grouped events**
> IMO, Title should be either Grouped entities or Grouped events. Depending on the use case Sub title: Loading entities or Loading events

Skipping for now. There was no design so I took the freedom to do it that way. Can iterate on it when there are clear requirements.

> **4. Missing colon after `Geo`**

Added.

<img width="446" height="117" alt="Screenshot 2025-11-03 at 14 49 05" src="https://github.com/user-attachments/assets/7278e437-f91e-45d3-a962-6cedfb483bf0" />
<img width="439" height="118" alt="Screenshot 2025-11-03 at 14 49 11" src="https://github.com/user-attachments/assets/7ae41d17-e04d-4a08-9215-8301e64bf318" />

> **5. Investigation's storybook is broken**

Skipped for now because it's out of scope. This was introduced by the change in the IPs popover:
- https://github.com/elastic/kibana/pull/240299

I found the problem though: We're calling directly the useExpandableFlyout API from GraphInvestigation and the story breaks because the component needs the ExpandableFlyout context provider. I'll raise a new PR to refactor this and inject the callback from GraphVisualization to decouple the flyout stuff from our package. That will solve it.

> **6. Hover button is not centered when hovered**

This is not an issue to polish after the release. EUI updated the icons and caused breaking changes, like misaligning our icon. It was not part of the scope, but fixed it too:

<img width="492" height="107" alt="Screenshot 2025-11-03 at 11 36 33" src="https://github.com/user-attachments/assets/5eece105-26a9-4f5f-9bee-a53040bec98f" />
<img width="265" height="233" alt="Screenshot 2025-11-03 at 11 36 38" src="https://github.com/user-attachments/assets/58434080-ffe9-49f9-9d2f-13512b23c6f1" />

> **7. High contrast of the background in borealis dark theme. I didn't expect to see difference from the light theme**

This is not an issue in Kibana. GroupedItem matches the flyout panel background in dark mode, as shown in this screenshot. The problem seems to be Storybook's dark theme config. Won't fix for now.

| Storybook | Kibana |
|--------|--------|
| <img width="482" height="857" alt="Screenshot 2025-11-03 at 11 50 04" src="https://github.com/user-attachments/assets/f2232b59-58ff-46a1-9816-d23a2a4455b3" /> | <img width="483" height="750" alt="Screenshot 2025-11-03 at 11 49 36" src="https://github.com/user-attachments/assets/580a9a5b-d181-4292-a1fa-3f87c2043f8c" /> | 

> **8. Show event details doesn't fit for groups** 

This wasn't a requirement and it was confirmed now by @larissaoliveiraelastic that we'll keep the current labels where the nodes represent groups or single items.

> 🐞 Issue:
> Total amount of items is larger than actual amount disabled when grouped

Skipping because it's out of scope: Some alerts are duplicated and triggered by the same events. We should dedupe/amend in a separate PR because this one is large enough already

> nit: following EUI pagination guidelines
> Not showing paging when results are less than the first page
> Keeping user preference of page size to localstorage

Wasn't aware of this since Figma file is plenty of pagination controls with a small subset of documents. But added both behaviors to the PR

Besides all of this, I also fixed a small issue I found with loading skeletons in grouped item. They don't grow any more:

| Before | After |
|--------|--------|
| <img width="597" height="1614" alt="Screenshot 2025-11-03 at 14 51 44" src="https://github.com/user-attachments/assets/2b2a290a-3eee-4ac7-a50a-795ae7446ccc" /> | <img width="596" height="1609" alt="Screenshot 2025-11-03 at 14 51 57" src="https://github.com/user-attachments/assets/4aa378c2-876e-4e38-b6e5-755ef05447db" /> |

albertoblaz committed:
Fix flyout unit tests

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/358054)
* Commit: 3216a8cd75c54bee3f488bf90d71c9d66b7bfa76
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-238395)
* Kibana Serverless Image: `docker.elastic.co/kibana-ci/kibana-serverless:pr-238395-3216a8cd75c5`

### Failed CI Steps
* [FTR Configs #47](https://buildkite.com/elastic/kibana-pull-request/builds/358054#019a4e86-ca1a-49c7-8752-a4da66e02937)

### History
* :broken_heart: [Build #357780](https://buildkite.com/elastic/kibana-pull-request/builds/357780) failed a61bccd3e6de67414f3e26ce081eac1fffd4c41a
* :broken_heart: [Build #356530](https://buildkite.com/elastic/kibana-pull-request/builds/356530) failed 3f9e09ead6798dae895c57a60a497a9b9dae73ed
* :broken_heart: [Build #356408](https://buildkite.com/elastic/kibana-pull-request/builds/356408) failed 0f4974f12f46c3916cdac6a45b3dd040cdfbdd81
* :broken_heart: [Build #356318](https://buildkite.com/elastic/kibana-pull-request/builds/356318) failed e7ee19bf7cd59fc304233fc4f5e2120ac100bf27
* :green_heart: [Build #354936](https://buildkite.com/elastic/kibana-pull-request/builds/354936) succeeded 66185a217428d9fc513b044187fbb33eb3957a57

cc @albertoblaz

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/358054","number":358054,"commit":"3216a8cd75c54bee3f488bf90d71c9d66b7bfa76"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/357780","number":357780,"commit":"a61bccd3e6de67414f3e26ce081eac1fffd4c41a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/356530","number":356530,"commit":"3f9e09ead6798dae895c57a60a497a9b9dae73ed"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/356408","number":356408,"commit":"0f4974f12f46c3916cdac6a45b3dd040cdfbdd81"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/356318","number":356318,"commit":"e7ee19bf7cd59fc304233fc4f5e2120ac100bf27"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/354936","number":354936,"commit":"66185a217428d9fc513b044187fbb33eb3957a57"}],"number":358054}
buildkite-pr-comment-->

albertoblaz committed:
Merge branch 'main' into graph-cleanup

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/358463)
* Commit: a8b68408f8274995ae5ded23fb8e95487a670edd
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-238395)
* Kibana Serverless Image: `docker.elastic.co/kibana-ci/kibana-serverless:pr-238395-a8b68408f827`

### History
* :broken_heart: [Build #358054](https://buildkite.com/elastic/kibana-pull-request/builds/358054) failed 3216a8cd75c54bee3f488bf90d71c9d66b7bfa76
* :broken_heart: [Build #357780](https://buildkite.com/elastic/kibana-pull-request/builds/357780) failed a61bccd3e6de67414f3e26ce081eac1fffd4c41a
* :broken_heart: [Build #356530](https://buildkite.com/elastic/kibana-pull-request/builds/356530) failed 3f9e09ead6798dae895c57a60a497a9b9dae73ed
* :broken_heart: [Build #356408](https://buildkite.com/elastic/kibana-pull-request/builds/356408) failed 0f4974f12f46c3916cdac6a45b3dd040cdfbdd81
* :broken_heart: [Build #356318](https://buildkite.com/elastic/kibana-pull-request/builds/356318) failed e7ee19bf7cd59fc304233fc4f5e2120ac100bf27

cc @albertoblaz

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/358463","number":358463,"commit":"a8b68408f8274995ae5ded23fb8e95487a670edd"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/358054","number":358054,"commit":"3216a8cd75c54bee3f488bf90d71c9d66b7bfa76"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/357780","number":357780,"commit":"a61bccd3e6de67414f3e26ce081eac1fffd4c41a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/356530","number":356530,"commit":"3f9e09ead6798dae895c57a60a497a9b9dae73ed"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/356408","number":356408,"commit":"0f4974f12f46c3916cdac6a45b3dd040cdfbdd81"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/356318","number":356318,"commit":"e7ee19bf7cd59fc304233fc4f5e2120ac100bf27"}],"number":358463}
buildkite-pr-comment-->

albertoblaz committed:
Merge branch 'main' into graph-cleanup

albertoblaz committed:
Restore refresh button in EmptyBody

kfirpeled reviewed (commented):
Thank you for taking the time to apply the changes!

🐞  I found another issue in the navigation of grouped events and entities - when navigating back the page's state is not applied to the shown results

Checkout attached video:

https://github.com/user-attachments/assets/3183e81b-fe53-4034-a955-8ff41a5a8845

albertoblaz committed:
Add missing colon to Geo tag

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/358624)
* Commit: a06a93292f754b6cd66d935ee8ca2fcf638fcc2e
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-238395)
* Kibana Serverless Image: `docker.elastic.co/kibana-ci/kibana-serverless:pr-238395-a06a93292f75`

### Failed CI Steps
* [FTR Configs #6](https://buildkite.com/elastic/kibana-pull-request/builds/358624#019a54a0-e8f2-4a4b-a21d-df854feda0dc)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/358624#019a54a0-e8f2-4a4b-a21d-df854feda0dc) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/358624/jobs/019a54a0-e8f2-4a4b-a21d-df854feda0dc/artifacts/019a54c2-8868-428c-bdd5-708b0dbdc39d) FTR Configs #<span></span>6 / lens serverless - group 1 - subgroup 6 lens logsdb smoke testing functions support Scenarios with changing stream type LogsDB downgraded to regular data stream scenarios LogsDB stream with no additional stream/index and no host.name field should visualize a date histogram chart using a different date field

### History
* :green_heart: [Build #358463](https://buildkite.com/elastic/kibana-pull-request/builds/358463) succeeded a8b68408f8274995ae5ded23fb8e95487a670edd
* :broken_heart: [Build #358054](https://buildkite.com/elastic/kibana-pull-request/builds/358054) failed 3216a8cd75c54bee3f488bf90d71c9d66b7bfa76
* :broken_heart: [Build #357780](https://buildkite.com/elastic/kibana-pull-request/builds/357780) failed a61bccd3e6de67414f3e26ce081eac1fffd4c41a
* :broken_heart: [Build #356530](https://buildkite.com/elastic/kibana-pull-request/builds/356530) failed 3f9e09ead6798dae895c57a60a497a9b9dae73ed
* :broken_heart: [Build #356408](https://buildkite.com/elastic/kibana-pull-request/builds/356408) failed 0f4974f12f46c3916cdac6a45b3dd040cdfbdd81

cc @albertoblaz

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/358624","number":358624,"commit":"a06a93292f754b6cd66d935ee8ca2fcf638fcc2e"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/358463","number":358463,"commit":"a8b68408f8274995ae5ded23fb8e95487a670edd"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/358054","number":358054,"commit":"3216a8cd75c54bee3f488bf90d71c9d66b7bfa76"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/357780","number":357780,"commit":"a61bccd3e6de67414f3e26ce081eac1fffd4c41a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/356530","number":356530,"commit":"3f9e09ead6798dae895c57a60a497a9b9dae73ed"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/356408","number":356408,"commit":"0f4974f12f46c3916cdac6a45b3dd040cdfbdd81"}],"number":358624}
buildkite-pr-comment-->

albertoblaz committed:
Localize group flyout title correctly

albertoblaz committed:
Render group panel for single alert (incl. event)

albertoblaz committed:
Fix React key warning

albertoblaz committed:
Fix pagination sync issues

- Sync initial in-memory state with local storage
- Sync PaginationControls and state defaults

It also encapsulate all pagination state

albertoblaz committed:
Normalize IP and Country arrays

albertoblaz committed:
Merge branch 'main' into graph-cleanup

albertoblaz committed:
Fix i18n quick check

albertoblaz committed:
Update tests for show event action for groups

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/359055)
* Commit: 6263414232f3664e264cb94c5b57d9ff785ba822
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-238395)
* Kibana Serverless Image: `docker.elastic.co/kibana-ci/kibana-serverless:pr-238395-6263414232f3`

### History
* :broken_heart: [Build #358915](https://buildkite.com/elastic/kibana-pull-request/builds/358915) failed 94e07936b8e632ee3d9dbafa131c9f170ffdbf23
* :broken_heart: [Build #358897](https://buildkite.com/elastic/kibana-pull-request/builds/358897) failed 637a5b9cc69eb69715b50bf8b73b83c1915df949
* :yellow_heart: [Build #358624](https://buildkite.com/elastic/kibana-pull-request/builds/358624) was flaky a06a93292f754b6cd66d935ee8ca2fcf638fcc2e
* :green_heart: [Build #358463](https://buildkite.com/elastic/kibana-pull-request/builds/358463) succeeded a8b68408f8274995ae5ded23fb8e95487a670edd
* :broken_heart: [Build #358054](https://buildkite.com/elastic/kibana-pull-request/builds/358054) failed 3216a8cd75c54bee3f488bf90d71c9d66b7bfa76

cc @albertoblaz

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/359055","number":359055,"commit":"6263414232f3664e264cb94c5b57d9ff785ba822"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/358915","number":358915,"commit":"94e07936b8e632ee3d9dbafa131c9f170ffdbf23"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/358897","number":358897,"commit":"637a5b9cc69eb69715b50bf8b73b83c1915df949"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/358624","number":358624,"commit":"a06a93292f754b6cd66d935ee8ca2fcf638fcc2e"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/358463","number":358463,"commit":"a8b68408f8274995ae5ded23fb8e95487a670edd"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/358054","number":358054,"commit":"3216a8cd75c54bee3f488bf90d71c9d66b7bfa76"}],"number":359055}
buildkite-pr-comment-->
